File: examples/async/mexc_ws_public.py
"""Subscribe to a MEXC public WebSocket stream."""

import asyncio

from dcex.ws import mexc


async def main() -> None:
    """Subscribe to MEXC spot trades and print the subscription response."""
    async with mexc.public() as ws:
        await ws.subscribe_trades("BTC-USDT-SPOT")
        print(await ws.recv())


if __name__ == "__main__":
    asyncio.run(main())

File: crates/dcex/examples/mexc_ws_public.rs
use std::time::Duration;

use dcex::ws::mexc::MexcPublicWebSocket;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let mut ws = MexcPublicWebSocket::new(Duration::from_secs(10))?;
    ws.connect().await?;
    ws.subscribe_trades("BTC-USDT-SPOT", "100ms").await?;
    println!("{}", String::from_utf8_lossy(&ws.recv().await?));
    ws.close().await?;
    Ok(())
}
